### PR TITLE
Replacing MaxEnt with MuonMaxent in the frequency domain analysis GUI

### LINKED ISSUE #21765: MuonMaxEnt in frequency domain analysis
muon group wanted the muonmaxent algorithm to replace the current maxent in the frequency domain analysis gui.

### PR DESCRIPTION
**Pull request by AnthonyLim23**
The frequency domain analysis (FDA) GUI was requested to fit a single frequency spectrum to all of the time domain spectra. So we have used the replaced MaxEnt with the MuonMaxent algorithm. 

**To test:**
1. Open muon analysis and load some data (MUSR00062260 is nice)
2. Open the frequency domain analysis (you can then close muon analysis)
3. Switch to the MaxEnt page by changing the top drop down menu
4. Click the Calculate button - and a frequency spectrum will be added to the grouped workspace
5. If you select the outputs then these will also be added to the grouped workspace with appropriate names
6. Select Use Phase Table and try to run it (it should fail and give an error message)
7. Change the workspace to the run number (instead of MuonAnalysisGrouped_1) and run the code
8. A table workspace (PhaseTable) should now exist. Open it and change the first phase to 5.
9. In the GUI untick the Construct Phase Table option
10. Tick the Fix phases and the Output phase table options
11. Click the calculate button
12. Check the output phase table has the same (changed) values as the original phase table (i.e. is the first phase 5).

Fixes #21765. 

**Release Notes** 

added to release notes and doc updated
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] Do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/interfaces/Frequency Domain Analysis.rst [hunk 1]
```diff
@@ -53,8 +53,12 @@ appropriate grouped workspace.
 Maximum Entropy Method
 ^^^^^^^^^^^^^^^^^^^^^^
 
-The maximum entropy method can be used to calculate the frequency domain spectrum. The MaxEnt Property table contains 
-the basic quantitites used during the calculation. The 
+The maximum entropy method can be used to calculate the frequency domain spectrum. This interface uses the :ref:`MuonMaxent <algm-MuonMaxent>` algorithm to calculate the frequency spectrum. The MaxEnt Property table contains 
+the basic quantitites used during the calculation. If the ``Use Phase Table`` checkbox is set to true, then an input phase table will be used for the initial phase values (the phase table will be called ``PhaseTable``). To generate
+an estimate phase table the ``Construct Phase Table`` option can be used and when the ``Calculate MaxEnt`` button is pressed ref:`CalMuonDetectorPhases <algm-CalMuonDetectorPhases>` is ran first to generate the phase table. If the
```
Comment (soininen) [inline docs/source/interfaces/Frequency Domain Analysis.rst]: Missing `:` in `ref:`.

**Review by soininen — CHANGES_REQUESTED** [review]
Tested the interface with some MUSR file found in the unit test data and everything seems to work as the instructions promise. The code looks good to me, as well. I spotted just a small error in the sphinx documentation, see below.

File: docs/source/interfaces/Frequency Domain Analysis.rst (hunk 1 above)
Comment (AnthonyLim23) [reply docs/source/interfaces/Frequency Domain Analysis.rst]: I will add the : before ref with the fix for the failing test

**Review by soininen — APPROVED** [review]

**Comment by martyngigg** [comment]
Unrelated build failure.

**Pull request merged**